aten/src/ATen/native/cuda/BinaryRemainderKernel.cu

```diff
@@ -37,17 +37,14 @@ void remainder_kernel_cuda(TensorIterator& iter) {
 }
 
 void fmod_kernel_cuda(TensorIterator& iter) {
-  // Use the dtype of the first argument to retain BC,
-  // change to common_dtype for type promotion in the future
-  // Issue #47779: https://github.com/pytorch/pytorch/issues/47779
-  if (isIntegralType(iter.dtype(), /*includeBool*/ false)) {
-    AT_DISPATCH_INTEGRAL_TYPES(iter.dtype(), "fmod_cuda", [&]() {
+  if (isIntegralType(iter.common_dtype(), /*includeBool*/ false)) {
+    AT_DISPATCH_INTEGRAL_TYPES(iter.common_dtype(), "fmod_cuda", [&]() {
       gpu_kernel_with_scalars(iter, []GPU_LAMBDA(scalar_t a, scalar_t b) -> scalar_t {
         return a % b;
       });
     });
   } else {
```

Comment:
Take this dispatch out of the else branch, like this:

```
if ... {
}

# dispatch
```

Reply:
I think we still need to keep it within the else block. Otherwise, we will dispatch two times for this operation.

Reply:
Oh I see what you're saying, yes that's fine. Sorry. Please ignore my suggestion to take them out, we can keep them in the branch.